src/main/kotlin/com/openlattice/organizations/DatasetController.kt

```diff
@@ -79,22 +79,41 @@ class DatasetController : DatasetApi, AuthorizingComponent {
 
     @Timed
     @GetMapping(path = [ID_PATH + EXTERNAL_DATABASE_TABLE + EXTERNAL_DATABASE_COLUMN])
-    override fun getExternalDatabaseTablesWithColumns(
-            @PathVariable(ID) organizationId: UUID): Map<OrganizationExternalDatabaseTable, Set<OrganizationExternalDatabaseColumn>> {
+    override fun getExternalDatabaseTablesWithColumnMetadata(
+            @PathVariable(ID) organizationId: UUID): Set<OrganizationExternalDatabaseTableColumnsPair> {
         val columnsByTable = edms.getExternalDatabaseTablesWithColumns(organizationId)
```

Comment:
This whole function body could be replaced with `getAuthorizedExternalDbTablesWithColumnMetadata(organizationId, Permission.READ)`

Reply:
I don't think so. The point of making two apis is because the customers want to be able to see "metadata" on any tables, so we need to return the column objects even if the person asking to see them don't have permissions on the column objects. It specifically doesn't do a check on column object authorizations.

The `getAuthorizedExternalDbTablesWithColumnMetadata` is so one can see what permissions they have across all the tables

Reply:
Oh.. you're right! I totally missed the filter on columnIds.. In this case, I would create a private function:
`getAuthorizedExternalDbTablesWithColumnMetadata(organizationId: UUID, permission: Permission, filterColumnIds: Boolean = true)` and use that, since there is a lot of code in common in these 2 functions